lib/virt_feature_test_base.pm

```diff
@@ -47,13 +47,70 @@ sub run_test {
     die('Please override this subroutine in children modules to run desired tests.');
 }
 
+sub prepare_run_test {
+    my $self = shift;
+
+    my $wait_script              = "600";
+    my $logs_cleanup_script_name = "clean_up_virt_logs.sh";
+    my $logs_cleanup_script_url  = data_url("virt_autotest/$logs_cleanup_script_name");
+
+    script_run("source /usr/share/qa/qa_test_virtualization/shared/standalone");
+    my $download_bash_script = "curl -s -o ~/$logs_cleanup_script_name $logs_cleanup_script_url";
+    script_output($download_bash_script, $wait_script, type_command => 0, proceed_on_failure => 0);
+    save_screenshot;
+    my $execute_bash_script = "chmod +x ~/$logs_cleanup_script_name && ~/$logs_cleanup_script_name -l \"/var/log/libvirt /var/lib/libvirt /var/log/xen\"";
+    script_output($execute_bash_script, $wait_script, type_command => 0, proceed_on_failure => 0);
+    save_screenshot;
+
+    $self->start_monitor_guest_console;
+}
+
+sub guest_console_monitor {
+    my ($self, $monitor_button) = @_;
+    my $monitor_option = "";
+
+    if ($monitor_button eq "start") {
+        $monitor_option = "-s";
+    }
+    elsif ($monitor_button eq "stop") {
+        $monitor_option = "-e";
+    }
+    else {
+        diag("Guest console monitor can only accept start or stop as options.");
+        return;
+    }
+
+    my $wait_script               = "180";
+    my $guest_console_script_name = "guest_console_monitor.sh";
+    my $guest_console_script_url  = data_url("virt_autotest/$guest_console_script_name");
+
+    my $download_bash_script = "curl -s -o ~/$guest_console_script_name $guest_console_script_url";
+    script_output($download_bash_script, $wait_script, type_command => 1, proceed_on_failure => 0);
+    save_screenshot;
+    my $execute_bash_script = "chmod +x ~/$guest_console_script_name && ~/$guest_console_script_name $monitor_option";
+    script_output($execute_bash_script, $wait_script, type_command => 1, proceed_on_failure => 0);
+    save_screenshot;
```

Comment:
Can you wrap this download/change mod|run script block to a function, so that it can be reused in function prepare_run_test/guest_console_monitor/etc?

Reply:
They are already wrapped in guest_console_monitor. And the function already served single purpose.

Reply:
Why is it not serving this feature base file as well as virt_auto base file?

Reply:
1. monitor_guest_console is already in virt_utils.pm. 
2. It is being used only here in feature test base file
3. I have no plan to use this monitor_geust_console in prj1-6 tests. Not very meaningful to use it there.

Reply:
I checked the monitor_guest_console function. It is not what I suggested in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/10850#discussion_r474438971. Function monitor_guest_console, although has "download/change mod|run script" code there, but it is not what "wrapping" is. By wrapping, I mean the regular function abstraction for only "download/change mod|run script" code. You should know, from the beginning, that it does not include guest console monitor related specific things -- just extract the common part into a single function, so  "download/change mod|run script" . Of course, no where else should reuse monitor_guest_console, instead the ~4 similar code blocks where download script from data and execute,  should call this common function.

Reply:
I do not think it is necessary to wrap download and execute together. There is no meaning in it. It should be left to the script function itself. There is no point to let all script functions call a command download/execute function. It is up to the script function itself to do their unique download and execution. This is better. Now you only see bash, there will be many other scripts in the future, download/execute function will decrease in its value along time elapse, although I do not think it has some value at the beginning. And I also mentioned many times, executing scripts are so diverse, there is no point to wrap execution and passing long trailing arguments again and improves nothing. By the way, you are suggesting to wrap something together serves no better purpose and situation. It just increases complexity, because I have to concern more about how to call your suggested download/execute function with careful thinking instead of doing my own things in script function. Obviously the latter is more practical and convenient. At the last, again today you only see bash scripts here, there will be many others in the future, do you need to implement another kind of down/execute for respective kind of script ? I do not think we to to that. There is no point to implement a standalone download/execute one. I see no meaning and purpose, just only for the sake of abstraction and unifying, and ignore totally about how to use them.

Reply:
Well, I have to say I do not agree mostly. Of course I can not force you to do it either.

Without a function for it,  with more and more scripts stored in openqa data directory in upcoming future, I can predict that there will be more and more such similar code blocks. @Julie-CAO HDYT?

Reply:
I know your point. I just express my own opinion, because I can not see convenience. It brings me no benefits. And I also consider that things will change and usage will be very diverse, the value of such implementation will diminish over time. And script function should have their own way of doing anything they are supposed to do instead of based another one that tells them what should be passed in.

Reply:
@Julie-CAO Can you help to have an evaluation for this?

Reply:
If it was used in two more places and with more than 3 lines to implement similar functions, I would like to create a common function to keep it neat and reusable.

Reply:
I might used too many variables which might not be quite necessary. Now I simplified them to only have two script_output to do the job. Please refer back to the updated code. Do you think it is still necessary to wrap them up ? I still do not think so. Please also refer to my opinions here: https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/10849#issuecomment-688014857
I think "it was used in two more places and with more than 3 lines to implement similar functions" is not quite enough because https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/10849#issuecomment-688014857

Reply:
I am trying to help, but I couldn't find the exact answer/guidance from Perl coding standards and article for how many lines duplication is better to be wrapped to function, however there are still something can be refers, fyi https://blog.csdn.net/yaoqiang2011/article/details/10232261#t45
I remember Travis checking tool will give the alert for those duplication codes. If you really hard to get the agreement of this coding style, follow Travis's check will be an option.